hca/added_command.py

```diff
@@ -207,27 +212,103 @@ def _build_body_payload(cls, namespace):
+            else:
+                logging.info("Found access token in kwargs.")
+                access_token = args['access_token']
+
+        # Checking config
+        elif config.get('access_token', None):
+            logging.info(config['access_token'])
+            # There is a refresh token
+            if retry and config.get('refresh_token', None):
+                logging.info("The access token stored in {} is not valid."
+                             " Attempting with refresh token.".format(config.config_files[-1]))
+                credentials = oauth2client.client.OAuth2Credentials(
+                    access_token=None,
+                    client_id=config.client_id,
+                    client_secret=config.client_secret,
+                    scopes=["https://www.googleapis.com/auth/userinfo.email"],
+                    refresh_token=config.refresh_token,
+                    token_expiry=None,
+                    token_uri=config.token_uri,
+                    user_agent=None,
+                    revoke_uri=None
+                )
+                # Silence ResourceWarning from httplib2 socket being open for connection pooling.
+                warnings.simplefilter("ignore")
+                credentials.refresh(httplib2.Http())
+
+                config.access_token = credentials.access_token
+                access_token = credentials.access_token
+            # No refresh token
+            elif retry:
+                logging.info("The access token stored in {} is not valid and there is"
+                             " no supplied refresh token.".format(config.config_files[-1]))
+                raise ValueError("The access_token in your config file is invalid"
+                                 " and there is no refresh_token to refresh it."
+                                 " You may run `hca login` to easily reset your"
+                                 " hca configuration.")
+            # First attempt
+            else:
+                logging.info("Found access token in {}.".format(config.config_files[-1]))
+                access_token = config.access_token
+
+        # Service account handling
+        elif 'GOOGLE_APPLICATION_CREDENTIALS' in os.environ:
+            logging.info("Found GOOGLE_APPLICATION_CREDENTIALS environment variable."
+                         " Grabbing service account credentials from there.")
+            service_account_credentials_filename = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+
+            if not os.path.isfile(service_account_credentials_filename):
+                raise EnvironmentError(
+                    "File {} (pointed by GOOGLE_APPLICATION_CREDENTIALS"
+                    " environment variable) does not exist!".format(service_account_credentials_filename))
+
+            service_account_class = oauth2client.service_account.ServiceAccountCredentials
+            service_account_credentials = service_account_class.from_json_keyfile_name(
```

Comment:
Using the google package directly now so this isn't as relevant, but I had seen setting a long class name this way in the argparse source code so I thought that could be nicer than wrapping the filename underneath. 

If the file is broken now the google auth package will handle it.